kernel/src/dynamic_process_loading.rs

```diff
@@ -0,0 +1,104 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+//! Dynamic Process Loader for application loading and updating at runtime.
+//!
+//! These functions facilitate dynamic application flashing and process creation
+//! during runtime without requiring the user to restart the device.
+
+use crate::config;
+use crate::debug;
+use crate::process;
+use crate::process::{ProcessLoadingAsync, ProcessLoadingAsyncClient};
+use crate::process_binary::ProcessBinaryError;
+use crate::process_loading::ProcessLoadError;
+use crate::utilities::cells::{MapCell, OptionalCell};
+use crate::ErrorCode;
+
+/// This interface supports loading processes at runtime.
+pub trait DynamicProcessLoading {
+    /// Call to request kernel to load a new process.
+    fn load(&self) -> Result<(), ErrorCode>;
+
+    /// Sets a client for the DynamicProcessLoading Object
+    ///
+    /// When the client operation is done, it calls the `load_done()`
+    /// function.
+    fn set_load_client(&self, client: &'static dyn DynamicProcessLoadingClient);
+}
```

Comment:
I think this trait is nice for separating concerns, but I don't think the `struct DynamicProcessLoader` is adding much in this implementation. I think `DynamicBinaryFlasher` should just implement `DynamicProcessLoading`.